tests/unit/ext/switches/brocade.py

```diff
@@ -0,0 +1,229 @@
+
+TRUNK_NATIVE_PAYLOAD = '<trunk><native-vlan>102</native-vlan></trunk>'
+
+TRUNK_VLAN_PAYLOAD = '<vlan><add>102</vlan></vlan>'
+
+TRUNK_REMOVE_VLAN_PAYLOAD = '<vlan><remove>102</remove></vlan>'
+
+INTERFACE1 = '104/0/10'
+INTERFACE2 = '104/0/18'
+INTERFACE3 = '104/0/20'
+
+
+class TestBrocade(object):
+    @pytest.fixture()
+    def switch(self):
+        return brocade.Brocade(
+            hostname='http://example.com',
+            username='admin',
+            password='admin',
+            interface_type='TenGigabitEthernet'
+        ).session()
+
+    def test_get_port_networks(self, switch):
+        with requests_mock.mock() as mock:
+            mock.get(switch._construct_url(INTERFACE1, suffix='trunk'),
+                     text=TRUNK_NATIVE_VLAN_RESPONSE_WITH_VLANS)
+            mock.get(switch._construct_url(INTERFACE2, suffix='trunk'),
+                     text=TRUNK_NATIVE_VLAN_RESPONSE_NO_VLANS)
+            mock.get(switch._construct_url(INTERFACE3, suffix='trunk'),
+                     text=TRUNK_VLAN_RESPONSE)
+            response = switch.get_port_networks([INTERFACE1,
+                                                 INTERFACE2,
+                                                 INTERFACE3])
+            assert response == {
+                INTERFACE1: [('vlan/native', '10'),
+                             ('vlan/4001', '4001'),
+                             ('vlan/4025', '4025')],
+                INTERFACE2: [('vlan/native', '10')],
+                INTERFACE3: [('vlan/1', '1'),
+                             ('vlan/4001', '4001'),
+                             ('vlan/4004', '4004'),
+                             ('vlan/4025', '4025'),
+                             ('vlan/4050', '4050')]
+            }
+
+    def test_get_mode(self, switch):
+        with requests_mock.mock() as mock:
+            mock.get(switch._construct_url(INTERFACE1, suffix='mode'),
+                     text=MODE_RESPONSE_ACCESS)
+            response = switch._get_mode(INTERFACE1)
+            assert response == 'access'
+
+            mock.get(switch._construct_url(INTERFACE1, suffix='mode'),
+                     text=MODE_RESPONSE_TRUNK)
+            response = switch._get_mode(INTERFACE1)
+            assert response == 'trunk'
+
+    def test_apply_networking(self, switch):
+        # Note(knikolla): Create a Nic connected to the Switch
+        port = model.Port(label=INTERFACE1, switch=switch)
+        nic = model.Nic(
+            model.Node(label='node-99',
+                       obm=Ipmi(type="http://schema.massopencloud.org/haas/v0/obm/ipmi",
+                                host="ipmihost",
+                                user="root",
+                                password="tapeworm")),
+            'ipmi',
+            '00:11:22:33:44:55')
+        nic.port = port
+        project = model.Project('anvil-nextgen')
+        network = model.Network(project, project, True, '102', 'hammernet')
```

Comment:
Could you put this stuff in a fixture/setup function? See my comments about the tests in #554 for rationale.